Kitodo/src/main/java/org/kitodo/production/interfaces/activemq/MapMessageObjectReader.java

```diff
@@ -110,6 +114,53 @@
         return mandatoryString;
     }
 
+    /**
+     * Fetches a {@code Collection<Integer>} from a MapMessage. This is a loose
+     * implementation for an optional object with optional content. The
+     * collection content is filtered through {@code toString()} and split on
+     * non-digits, dealing generously with list variants and separators. If not
+     * found, returns an empty collection, never {@code null}.
+     *
+     * @param key
+     *            the name of the set to return
+     * @return the set requested
+     * @throws JMSException
+     *             can be thrown by MapMessage.getObject(String)
+     */
+    public Collection<Integer> getCollectionOfInteger(String key) throws JMSException {
+        Collection<String> collectionOfString = getCollectionOfString(key);
+        List<Integer> collectionOfInteger = collectionOfString.stream()
+                .flatMap(string -> Arrays.stream(string.split("\\D+"))).filter(Strings::isNotBlank)
+                .map(Integer::valueOf).collect(Collectors.toList());
```

Comment:
False positive. Cannot throw NumberFormatException, because `string.split("\\D+")` will only return sequences of digits or empty strings, in case of non-digit characters at the beginning or end of the string. The latter are filtered out by `.filter(Strings::isNotBlank)`, thus leaving no chance for NumberFormatException to arise.

Reply:
Why not replace `.filter(Strings::isNotBlank)` with `.filter(StringUtils::isNumeric)`? That would take care of empty Strings and `values` as well (exactly what `isNotBlank` does), would perform an explicit number check and would be better readable and understandable, IMHO. 

It also would not require any new dependencies as we already use the Apache Commons library and the `StringUtils` class in multiple other places in the code.